libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -183,6 +182,14 @@ void AP_UAVCAN::init(uint8_t driver_index)
 
     hw_version.major = AP_UAVCAN_HW_VERS_MAJOR;
     hw_version.minor = AP_UAVCAN_HW_VERS_MINOR;
+
+    const uint8_t uid_buf_len = hw_version.unique_id.capacity();
+    uint8_t uid_len = uid_buf_len;
+    uint8_t unique_id[uid_buf_len];
+
+    hal.util->get_system_id_unformatted(unique_id, uid_len);
```

Comment:
shouldn't we do something if this returns false, for example on Linux? Or implement it on Linux based on hash of hostname?